status/types.proto

```diff
@@ -124,6 +124,9 @@ enum Object {
 
   // [**2053**] Invalid range has been requested for an object.
   OUT_OF_RANGE = 5;
+
+  // [**2054**] Size limit was exceeded, no further PUTs will be allowed.
+  SIZE_LIMIT_EXCEEDED = 6;
```

Comment:
QUOTA_EXCEEDED?

Reply:
it's general purpose to me, so i'd make it a `RESOURCE_EXHAUSTED` in `CommonFail`. By analogy with [gRPC](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) or kinda [HTTP](https://datatracker.ietf.org/doc/html/rfc6585#section-4)

Reply:
> RESOURCE_EXHAUSTED in CommonFail

BTW, HTTP example is a temporary error, retrying the same request in 10 minutes can succeed. Quotas are different, this error is permanent unless either quota is changed or some other objects are deleted.